tests/hp/hp_line_dof_identities_mixed_multiway.cc

```diff
@@ -0,0 +1,78 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2005 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// check FE_Collection::hp_line_dof_identities with multiway
+// identities for a combination of FE_Q element and FE_DGQ
+// elements. Only the FE_Q elements have DoFs on vertices so only they
+// will participate in the identities.
+
+
+#include <deal.II/fe/fe_dgq.h>
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/hp/fe_collection.h>
+
+#include "../tests.h"
+
+
+
+template <int dim>
+void
+test()
+{
+  deallog << "dim=" << dim << std::endl;
+
+  hp::FECollection<dim> fe_collection;
+  for (unsigned int i = 1; i < 8; ++i)
+    {
+      fe_collection.push_back(FE_Q<dim>(i));
+      fe_collection.push_back(FE_DGQ<dim>(i));
+    }
+
+  // construct the complete set of fe indices
+  std::set<unsigned int> fe_indices;
+  for (unsigned int i = 0; i < fe_collection.size(); ++i)
+    fe_indices.emplace(i);
+
+  const auto identities = fe_collection.hp_line_dof_identities(fe_indices);
+
+  for (unsigned int i = 0; i < identities.size(); ++i)
+    {
+      deallog << "Identity set #" << i << std::endl;
+      for (const auto &p : identities[i])
+        {
+          deallog << "  " << fe_collection[p.first].get_name()
+                  << ": line dof index " << p.second << std::endl;
```

Comment:
```suggestion
      deallog << "Identity set #" << i;
      for (const auto &p : identities[i])
        {
          deallog << std::endl << "  " << fe_collection[p.first].get_name()
                  << ": line dof index " << p.second;
```
Here and in the following: You can remove the blank line in the output file by writing it like this, but this is really not important.